plugins/module_utils/network/sonic/config/snmp/snmp.py

```diff
@@ -0,0 +1,1356 @@
+        elif state == 'merged':
+            commands, requests = self._state_merged(want, have)
+
+        return commands, requests
+
+    def new_test_keys(self, want):
+        """ Returns the test keys based on if want contains name value for agentaddress
+        and/or host options
+        :rtype: A list
+        :returns: new test_keys
+        """
+        test_keys = [
+            {'community': {'name': ''}},
+            {'group': {'name': '', 'access': {'security_model': ''}}},
+            {'engine': ''},
+            {'access': {'security_model': ''}},
+            {'user': {'name': ''}},
+            {'view': {'name': ''}}]
+
+        if 'agentaddress' in want:
+            no_name = False
+            for command in want['agentaddress']:
+                if 'name' not in command:
+                    no_name = True
+            if no_name:
+                test_keys.extend([{'agentaddress': {'ip': ''}},
+                                  {'agentaddress': {'vrf': ''}}])
+            else:
+                test_keys.extend([{'agentaddress': {'name': ''}}])
+        if 'host' in want:
+            no_name = False
+            for command in want['host']:
+                if 'name' not in command:
+                    no_name = True
+            if no_name:
+                test_keys.extend([{'host': {'ip': ''}},
+                                  {'host': {'vrf': ''}}])
+            else:
+                test_keys.extend([{'host': {'name': ''}}])
+
+        return test_keys
+
+    def _state_replaced_or_overridden(self, want, have, state):
+        """
+        The command generator when state is overridden
+        :rtype: A list
+        :returns: the commands necessary to migrate the current configuration
+                  to the desired configuration
+        """
+        commands = []
+        requests = []
+
+        if not want:
+            return commands, requests
+
+        diff_want = get_diff(want, have, TEST_KEYS)
+        del_commands = get_diff(have, want, TEST_KEYS)
+        if not diff_want and not del_commands:
+            return commands, requests
+        merged_commands = None
+        merged_request = None
+
+        if del_commands:
+            if 'agentaddress' in del_commands and 'agentaddress' in want:
+                for command in del_commands.get('agentaddress'):
+                    if command.get('name'):
+                        del_commands['agentaddress'].remove(command)
+                if len(del_commands['agentaddress']) == 0:
+                    del_commands.pop('agentaddress')
+            if 'host' in del_commands and 'host' in want:
+                for command in del_commands.get('host'):
+                    if command.get('name'):
+                        del_commands['host'].remove(command)
+                if len(del_commands['host']) == 0:
+                    del_commands.pop('host')
+            if len(del_commands) > 0:
```

Comment:
Before this final part of the function, an "else" block is needed to handle request generation for options in "diff_want" that do not correspond to any options that were previously configured (new option configuration to add, even though no deletions were needed).